### PR TITLE
Add Trino 442 release notes

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 442 release.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 14 Mar 2024

* #21059 ✅ rn ✅ docs
* #21063 ✅ rn ✅ docs
* #21070 ✅ rn ✅ docs
* #21057 ✅ rn ✅ docs
* #21067 ✅ rn ✅ docs
* #21030 ✅ rn ✅ docs
* #21080 ✅ rn ✅ docs
* #21079 ✅ rn ✅ docs
* #21054 ✅ rn ✅ docs
* #21074 ✅ rn ✅ docs
* #21082 ✅ rn ✅ docs
* #21062 ✅ rn ✅ docs
* #21086 
* #21092

### CODE REVIEW

File: docs/src/main/sphinx/release/release-440.md [hunk 1]
```diff
@@ -63,6 +63,10 @@
 
 ## Iceberg connector
 
+```{warning}
+This release has a major regression which is fixed in Release 442.
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-440.md:67]:
```suggestion
This release has a major regression which is fixed in Trino 442.
```

also do we need the same in 441? 

I am not sure we need this at all .. did we do this in the past?

File: docs/src/main/sphinx/release/release-442.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Release 442 (14 Mar 2024)
+
+## Delta Lake connector
+
+* Fix query failure when a partition value contains forward slash characters. ({issue}`21030`)
+
+## Iceberg connector
+
+* Fix `NullPointerException` query failure. ({issue}`21074`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-442.md:9]: Do we need to add info when that happens?

File: docs/src/main/sphinx/release/release-442.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+## OpenSearch connector
+
+* Add support for configuring AWS deployment type with the
+  `opensearch.aws.deployment-type` configuration property. ({issue}`21059`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-442.md:14]: I hope thats documented...

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-440.md (hunk 1 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-440.md:67]: Yeah, we've added a warning for major regressions that required emergency releases to patch. I don't think we need it in 441 though

File: docs/src/main/sphinx/release/release-442.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-442.md:9]: It's during split enumeration, which I don't know how to frame in a user-friendly way?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-442.md:9]: maybe jsut say 

Prevent NullPointerException from split enumeration during query processing

Comment (mosabua) [reply docs/src/main/sphinx/release/release-442.md:9]: Its technical but at least its not vague

**Pull request merged**